plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -584,24 +616,62 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
-    public static LongWriteFunction trinoTimestampToOracleTimestampWriteFunction()
+    private static ObjectWriteFunction oracleLongTimestampWriteFunction(TimestampType timestampType)
+    {
+        int roundToPrecision = timestampType.getPrecision();
+        checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        checkArgument(
+                6 <= roundToPrecision && roundToPrecision <= 9 && roundToPrecision <= timestampType.getPrecision(),
```

Comment:
redundant check `roundToPrecision <= timestampType.getPrecision()`? As `roundToPrecision` is `timestampType.getPrecision()`

Reply:
refactored